micro-rdk/src/common/data_manager.rs

```diff
@@ -51,6 +86,24 @@ where
         })
     }
 
+    pub fn from_robot_and_config(
```

Comment:
I'll admit I have some worries about the fact that the service isn't being created via the same configuration parsing and instantiation logic that works with the regular robot config. If the `DataManagerService` were to become a more active component, in the sense of having inbound-to-robot RPCs that we needed to honor, how would those get acted on?

I'm willing to handwave it for now I think, and let the data manager be a one-off, but if we ever wanted to implement another service, I think at that point I'd want to see the micro-rdk grow a real service concept and then fold the data service in under that mechanism.

Also worth noting that currently, there isn't any real difference between a component and a service, nor does micro-rdk have a concept of resource/component/service that I'm aware of.

In any event, I'm not proposing we make changes now - just pointing it out as future work.

Reply:
Just because it lacks a robust outward facing api today, for our purposes, doesn't mean there won't be one in the future, even if our means of transporting data is push vs pull, there could be other functionality added later. I'm hopeful that when we need to add the next service, whatever it is, that we can find a way to pull the data stuff into that framework. But, it isn't something to worry about now, anyway.